### ISSUE TITLE
Unintended "slide-in from bottom" activation of ProSwitcher/LibActivator

### ISSUE THREAD

**Issue by PrabhuDeva** [rpetrich/libactivator#17]
I have moved my topic (including Ryan's first reply and then my reply) here from the ProSwitcher site, because it's really meant to be here and I think it deserves some more discussion...

---

I really love the "slide-in from bottom" activation methods provided by LibActivator as they're a great way of activating ProSwitcher without blocking home button double click for the iPod-popup.

However, it seems to be far too sensitive. For example, when scrolling through a website in landscape Safari (where my finger slides over the right edge of the touchscreen, directly above the home button), ProSwitcher is activated very often which is unintended. Apparently, the slide-in gesture is already activated when only sliding ones finger slightly below the Dock icons. I think one should have to slide one finger as high as the Springboard page indicator dots in order to avoid unintended activation.

Apart from that, great work guys!!

---

rpetrich about 9 hours ago

It only counts when you drag onto the screen from outside, not from inside the screen past the edge. You are probably inadvertently dragging onto the screen.
Comment
It only counts when you drag onto the screen from outside, not from inside the screen past the edge. You are probably inadvertently dragging onto the screen.

---

PrabhuDeva 4 minutes ago

Yes, that's what you'd think I do. But in fact, I'm not. When holding Safari in landscape, home button on the right side, I slide my thumb up and down (only vertically, so no inadvertent dragging my finger onto the screen physically!) on the rightmost possible side of the screen in order not to block stuff with my thumb. Then very often, one of the three slide-in gestures is triggered if activated. As far as I know, the multitouch screen has quite a hard time properly tracking finger movement at the very edge of the screen, so the problem might be due to Apple's touch detection system. I really can't imagine I'm the only one with this problem. I've been experimenting with having sleep button on the left slide-in gesture, ProSwitcher on the middle slide-in, and home button on the right slide-in (in which case one would be able to operate the iPhone without ever having to press a button apart from recovering from sleep mode). But having sleep or home button on the slide-in gestures is really "dangerous" due to inadvertently closing apps or putting the device to sleep mode, which really is not nice. Basically, this problem for me really reduces the huge (!) potential of using LibActivator's slide-in gestures for a wide range of system or app functionalities.

Basically, I'd love it if you'd either increase the distance one's finger has to travel from outside the screen towards the middle of the screen before triggering the slide-in gestures, or maybe even having an option of setting one's own preferred "sensitivity". I think it really would not matter for people having to slide their finger further before triggering (even if they don't currently experience my problem) since it's a difference of milliseconds. But again, in my opinion this would really increase the huge potential of using LibActivator's slide-in gestures more "safely"...

Thanks for bothering, Ryan!!

**Comment by rpetrich** [comment]
I think I have a potential solution... right now I'm detecting any touch that starts on the edge of the screen as a swipe in. What I should do is flag those touches as potential swipe ins, but only of they continue towards the center of the screen.

You're right, the touch events aren't very accurate near the edges of the screen and the bottom edge is actually the worst of all (at least on the three devices I tested personally)--some heuristics are probably necessary

This won't make it into the public 1.2 release which has already been submitted

**Comment by PrabhuDeva** [comment]
Okay, great. I'll just wait whether you'll come up with a solution to that problem.

By the way, these unintentional activations of the slide-in gestures also do happen quite a lot when you're writing a mail or a text message in landscape (again, home button on the right side), then go to the "add recipient" list and use the alphabetical scroll bar on the side. I'm sure you can reproduce that yourself.

**Comment by rpetrich** [comment]
I have merged this into the latest beta. Please give it a try

**Comment by PrabhuDeva** [comment]
Uhm, I'd like to... but where to find this beta? It's not in Cydia, is it?

**Comment by rpetrich** [comment]
It's in Cydia, but you have to add the following repo: http://rpetri.ch/repo

**Comment by PrabhuDeva** [comment]
Thanks, I didn't know about that beta repo.

And, let me say, great work! As far as I've tested it, it does work like a charm now! Hope you're happy with it, too...

So thanks again. In my opinion, Activator is one of the best overall improvements to the iPhone UI via a jailbreak tweak there is, if not the best of all -- especially given the growing degree of integration with other apps.
